src/partition/repart.c

```diff
@@ -4865,6 +5049,32 @@ static int add_exclude_path(const char *path, Hashmap **denylist, DenyType type)
         return 0;
 }
 
+static int shallow_join_strv(char ***ret, char **a, char **b) {
+        _cleanup_free_ char **joined = NULL;
+        char **iter;
+
+        assert(ret);
+
+        joined = new(char*, strv_length(a) + strv_length(b) + 1);
+        if (!joined)
+                return log_oom();
+
+        iter = joined;
+
+        STRV_FOREACH(i, a)
+                *(iter++) = *i;
+
+        STRV_FOREACH(i, b)
+                *(iter++) = *i;
+
+        *iter = NULL;
+
+        strv_uniq(joined);
```

Comment:
doesn't matter, but maybe do this once after both shallow_join_strv() calls, since kinda slow

Reply:
There's no situation where we're calling shallow_join_strv twice on one array.

BUT, what I'm doing here is outright wrong. `strv_uniq` frees the strings in the strv as it removes them, but since this is a shallow strv join we don't actually want to free them (otherwise we'll get double-free...)

Reply:
Alright reworked a different way that shouldn't cause bugs. PTAL